### PR TITLE
Revise Sort Docs

### PR DESCRIPTION
**Pull request by kanitw**
Merge #3944 First

### CODE REVIEW

File: site/docs/encoding/sort.md [hunk 1]
```diff
@@ -38,28 +45,33 @@ Setting x's `sort` to `"descending"` reverses the x-axis. Thus, the following vi
 <div class="vl-example" data-name="tick_sort"></div>
 
 
-## Sorting Discrete Scales
+## Sorting Discrete Fields
 
-If the channel has a discrete scale (`band`, `point` or `ordinal`), the field's values of the channel can be sorted in the following ways:
+If the channel has a discrete scale (ordinal or nominal), `sort` can be one of the following: `"ascending"`, `"descending"`, [a sort field definition](#sort-field) for sorting by another field, [an array specifying preferred order](#sort), or `null`.
```
Comment (domoritz) [inline site/docs/encoding/sort.md]: can be one of: (no need to say the following)

File: site/docs/encoding/sort.md [hunk 2]
```diff
@@ -38,28 +45,33 @@ Setting x's `sort` to `"descending"` reverses the x-axis. Thus, the following vi
...
 
+### Sort by a Different Field
 {:#sort-field}
 
 2) Sorting by aggregated value of another "sort" field. In this case, `sort` is an __encoding sort field definition__, which has the following properties:
 
 {% include table.html props="field,op,order" source="EncodingSortField" %}
 
-3) Unsorted – `null` – The field is not sorted. This is equivalent to specifying `sort: false` in [Vega's scales](https://vega.github.io/vega/docs/scales/#sort).
+For example, the following plot sorts x by mean of Horsepower.
```
Comment (domoritz) [inline site/docs/encoding/sort.md]: sorts by the mean

File: site/docs/encoding/sort.md [hunk 3]
```diff
@@ -38,28 +45,33 @@ Setting x's `sort` to `"descending"` reverses the x-axis. Thus, the following vi
...
-{:#sort-array}
+<div class="vl-example" data-name="histogram_sort_mean"></div>
 
-4) Sorting by preferred order of values by providing `sort` as an array of values that specify the order. Unspecified values will assume their original orders in the data source, preceded by the orders in the sort array.
 
-__Note__: It is also possible to sort by providing custom `scale`'s [`domain`](scale.html#domain). However, it is more error-prone compared to using `sort` array since `domain` requires every possible value to be included in the array. Thus, any value omitted from `domain` will not render properly.
+### Specifying Custom Sort Order
+{:#sort-array}
+
+If the `sort` property is an array, it specifies the preferred order of values.
```
Comment (domoritz) [inline site/docs/encoding/sort.md]: replace , with .

File: site/docs/encoding/sort.md [hunk 4]
```diff
@@ -69,11 +81,22 @@ If some values are ignored, the sort order will precede by the specified values
 
 <div class="vl-example" data-name="bar_custom_sort_partial"></div>
 
-#### Example: Sorting Ordinal Scale by Another Field
+For discrete time field, values in the sort array can be [date-time definition objects](types#datetime).
```
Comment (domoritz) [inline site/docs/encoding/sort.md]: for discrete time fieldS

File: site/docs/encoding/sort.md [hunk 5]
```diff
@@ -69,11 +81,22 @@ If some values are ignored, the sort order will precede by the specified values
...
+In addition, for time units `"month"` and `"day"`, the values can be the month or day names (case insensitive) or their 3-letter initials (e.g., `"Mon"`, `"Tue"`).
 
-The following example sorts x by mean of Horsepower.
+For example, the following chart orders the day to start on Monday (instead of Sunday by default).
 
-<div class="vl-example" data-name="histogram_sort_mean"></div>
+<div class="vl-example" data-name="circle_github_punchcard"></div>
+
+
+__Note__: It is also possible to sort by providing custom `scale`'s [`domain`](scale.html#domain). However, it is more error-prone compared to using `sort` array since `domain` requires every possible value to be included in the array. Thus, any value omitted from `domain` will not render properly.
```
Comment (domoritz) [inline site/docs/encoding/sort.md]: it is possible to sort by providing a custom scale domain

Comment (domoritz) [reply site/docs/encoding/sort.md]: compared to using A sort array

File: site/docs/encoding/sort.md (hunk 3 above)
Comment (kanitw) [reply site/docs/encoding/sort.md]: huh?

**Pull request merged**